### PR TITLE
SQL 1 exercise

### PR DESCRIPTION
**Pull request by NoyRobinson**
SQL 1 exercise

### CODE REVIEW

File: src/main/resources/db/migration/V202409081400__create-owner.sql [hunk 1]
```diff
@@ -0,0 +1,10 @@
+CREATE TABLE owner
+(
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    name varchar(255),
+    company_id UUID,
+    employee_id UUID
+);
+
+CREATE INDEX idx_owner_company_id ON owner(company_id);
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081400__create-owner.sql]: create one index for both

File: src/main/resources/db/migration/V202409081401__create-pet.sql [hunk 2]
```diff
@@ -0,0 +1,31 @@
+CREATE TABLE pets
+(
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    name varchar(255),
+    type varchar(255),
+    company_id UUID,
+    date_of_arrival DATE
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081401__create-pet.sql]: you can add here default value

File: src/main/resources/db/migration/V202409081400__create-owner.sql [hunk 3]
```diff
@@ -0,0 +1,10 @@
+CREATE TABLE owner
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081400__create-owner.sql]: the migration file name should be separated with '_' and not '-'

File: src/main/resources/db/migration/V202409081401__create-pet.sql [hunk 4]
```diff
@@ -0,0 +1,31 @@
+CREATE TABLE pets
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081401__create-pet.sql]: the migration file name should be separated with '_' and not '-'

File: src/main/resources/db/migration/V202409081401__create-pet.sql [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+-- VALUES ('Spot', 'rabbit', '50fcdfda-b823-47e6-9bad-541e87163c9a', '2024-09-01');
+
+-- INSERT INTO pets (name, type, company_id, date_of_arrival)
+-- VALUES ('Charlie', 'bird', '8ba5f6aa-b0d8-45b8-93a3-78429bcb1ee5', '2024-09-08');
+
+-- SELECT * FROM pets WHERE type = 'dog'
+
+-- DELETE FROM pets WHERE id = '98d200c4-fad5-4980-b72a-23f08f27bf63';
+
+-- SELECT * FROM pets WHERE date_of_arrival < '2023-09-08';
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081401__create-pet.sql]: this is not the query - Get the pets that arrived more than a year ago
this query brings all pets that arrived before '09-08-2023'

File: src/main/resources/db/migration/V202409081401__create-pet.sql [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+-- INSERT INTO pets (name, type, company_id, date_of_arrival)
+-- VALUES ('Max', 'cat', '50fcdfda-b823-47e6-9bad-541e87163c9a', '2010-05-10');
+
+-- INSERT INTO pets (name, type, company_id, date_of_arrival)
+-- VALUES ('Spot', 'rabbit', '50fcdfda-b823-47e6-9bad-541e87163c9a', '2024-09-01');
+
+-- INSERT INTO pets (name, type, company_id, date_of_arrival)
+-- VALUES ('Charlie', 'bird', '8ba5f6aa-b0d8-45b8-93a3-78429bcb1ee5', '2024-09-08');
+
+-- SELECT * FROM pets WHERE type = 'dog'
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081401__create-pet.sql]: you can use distinct

**Review by mayhuberman — CHANGES_REQUESTED** [review]
👑

File: src/main/resources/db/migration/V202409081400__create_owner.sql [hunk 7]
```diff
@@ -0,0 +1,9 @@
+CREATE TABLE owner
+(
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    name varchar(255),
+    company_id UUID,
+    employee_id UUID
+);
+
+CREATE INDEX idx_owner ON owner(company_id, employee_id);
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081400__create_owner.sql]: add the fields names to the index name

File: src/main/resources/db/migration/V202409081400__create_owner.sql [hunk 8]
```diff
@@ -0,0 +1,9 @@
+CREATE TABLE owner
+(
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    name varchar(255),
+    company_id UUID,
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081400__create_owner.sql]: company_id in bob is bigint

File: src/main/resources/db/migration/V202409081400__create_owner.sql [hunk 9]
```diff
@@ -0,0 +1,9 @@
...
+    employee_id UUID
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081400__create_owner.sql]: employee_id in bob is string

File: src/main/resources/db/migration/V202409081401__create_pet.sql [hunk 10]
```diff
@@ -0,0 +1,31 @@
+CREATE TABLE pets
+(
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    name varchar(255),
+    type varchar(255),
+    company_id UUID,
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081401__create_pet.sql]: company_id in bob is bigint

File: src/main/resources/db/migration/V202409081401__create_pet.sql [hunk 11]
```diff
@@ -0,0 +1,31 @@
+CREATE TABLE pets
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081401__create_pet.sql]: By the way, it's better to be consistent, and the name should be 'pet' in the singular

File: src/main/resources/db/migration/V202409081400__create_owner.sql [hunk 12]
```diff
@@ -0,0 +1,9 @@
+CREATE TABLE owner
+(
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    name varchar(255),
+    company_id BIGINT,
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081400__create_owner.sql]: missing not null

File: src/main/resources/db/migration/V202409081400__create_owner.sql [hunk 13]
```diff
@@ -0,0 +1,9 @@
...
+    employee_id String
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081400__create_owner.sql]: should be varchar and missing not null

File: src/main/resources/db/migration/V202409081401__create_pet.sql [hunk 14]
```diff
@@ -0,0 +1,31 @@
+CREATE TABLE pet
+(
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    name varchar(255),
+    type varchar(255),
+    company_id BIGINT,
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081401__create_pet.sql]: should be not null

File: src/main/resources/db/migration/V202409081400__create_owner.sql (hunk 12 above)
Comment (mayhuberman) [reply src/main/resources/db/migration/V202409081400__create_owner.sql]: think what other fields shouldn't hold null values

File: src/main/resources/db/migration/V202409081401__create_pet.sql [hunk 15]
```diff
@@ -0,0 +1,31 @@
+CREATE TABLE pet
+(
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    name varchar(255) NOT NULL ,
+    type varchar(255) NOT NULL ,
+    company_id BIGINT NOT NULL ,
+    date_of_arrival DATE DEFAULT current_date
+);
+
+CREATE INDEX idx_pets_company_id ON pets(company_id);
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081401__create_pet.sql]: it wont work because you renamed the table to pet

Comment (NoyRobinson) [reply src/main/resources/db/migration/V202409081401__create_pet.sql]: you're right, I'll change it

File: src/main/resources/db/migration/V202409081401__create_pet.sql [hunk 16]
```diff
@@ -0,0 +1,31 @@
+CREATE TABLE pet
+(
+    id UUID PRIMARY KEY DEFAULT gen_random_uuid(),
+    name varchar(255) NOT NULL ,
+    type varchar(255) NOT NULL ,
+    company_id BIGINT NOT NULL ,
+    date_of_arrival DATE DEFAULT current_date
+);
+
+CREATE INDEX idx_pets_company_id ON pet(company_id);
```
Comment (mayhuberman) [inline src/main/resources/db/migration/V202409081401__create_pet.sql]: better rename it to idx_pet_on_company_id

**Review by mayhuberman — APPROVED** [review]

**Pull request merged**